core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -0,0 +1,112 @@
+  * limitations under the License.
+  */
+package kafka.server
+
+import java.util.Properties
+
+import kafka.admin.AdminUtils
+import kafka.log.LogConfig
+import kafka.metrics.KafkaMetricsGroup
+import kafka.utils._
+import org.apache.kafka.common.errors.InvalidRequestException
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.requests.CreateTopicsRequest._
+
+import scala.collection._
+import scala.collection.JavaConverters._
+
+class AdminManager(val config: KafkaConfig,
+                   val metrics: Metrics,
+                   val metadataCache: MetadataCache,
+                   val zkUtils: ZkUtils) extends Logging with KafkaMetricsGroup {
+  this.logIdent = "[Admin Manager on Broker " + config.brokerId + "]: "
+
+  val topicPurgatory = DelayedOperationPurgatory[DelayedOperation]("topic", config.brokerId)
+
+  /**
+    * Try to complete delayed topic operations with the request key
+    */
+  def tryCompleteDelayedTopicOperations(topic: String) {
+    val key = TopicKey(topic)
+    val completed = topicPurgatory.checkAndComplete(key)
+    debug(s"Request key ${key.keyLabel} unblocked $completed topic requests.")
+  }
+
+  /**
+    * Create topics and wait until the topics have been completely created.
+    * The callback function will be triggered either when timeout, error or the topics are created.
+    */
+  def createTopics(timeout: Int,
+                   createInfo: Map[String, TopicDetails],
+                   responseCallback: Map[String, Errors] => Unit) {
+
+    // 1. map over topics creating assignment and calling zookeeper
+    val brokers = metadataCache.getAliveBrokers.map { b => kafka.admin.BrokerMetadata(b.id, b.rack) }
+    val metadata = createInfo.map { case (topic, arguments) =>
+      try {
+        val configs = new Properties()
+        arguments.configs.asScala.foreach { case (key, value) =>
+          configs.setProperty(key, value)
+        }
+        LogConfig.validate(configs)
+
+        val assignments = {
+          if ((arguments.numPartitions != NO_NUM_PARTITIONS || arguments.replicationFactor != NO_REPLICATION_FACTOR)
+            && !arguments.replicasAssignments.isEmpty)
+            throw new InvalidRequestException("Both numPartitions or replicationFactor and replicasAssignments were set. " +
+              "Both cannot be used at the same time.")
+          else if (!arguments.replicasAssignments.isEmpty) {
+            // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
+            // this follows the existing logic in TopicCommand
+            arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
+              (partitionId.intValue, replicas.asScala.map(_.intValue))
+            }
+          } else {
+            AdminUtils.assignReplicasToBrokers(brokers, arguments.numPartitions, arguments.replicationFactor)
```

Comment:
In the case that only numPartitions or only replicationFactor is defined, that will hit the else statement and call AdminUtils.assignReplicasToBrokers. That function will handle and check its parameters resulting in an exception if only one is defined. It also handles things like checking if the replication factor is higher than the number of brokers. 

The test CreateTopicsRequestsTest.testErrorCreateTopicsRequests validates the behavior.